frontend/kubecloud/src/utils/userService.ts

```diff
@@ -157,6 +157,24 @@ export class UserService {
     })
   }
 
+  // Fetch all grid nodes from the new public endpoint (no auth required)
+  async listAllGridNodes(filters?: { healthy?: boolean; size?: number; page?: number }) {
+    const queryParams = new URLSearchParams()
+    if (filters) {
+      Object.entries(filters).forEach(([key, value]) => {
+        if (value !== undefined && value !== null) {
+          queryParams.append(key, String(value))
+        }
+      })
+    }
+
+    const endpoint = `/v1/nodes${queryParams.toString() ? `?${queryParams.toString()}` : ''}`;
+    return api.get<NodesResponse>(endpoint, {
+      requiresAuth: false,
+      showNotifications: false // Don't show notifications for node listing
+    })
+  }
+
```

Comment:
I think this should be in statusService or node manager, maybe, but it is not related to the user at all imo